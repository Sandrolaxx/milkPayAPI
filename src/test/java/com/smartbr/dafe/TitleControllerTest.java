package com.smartbr.dafe;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import com.google.inject.Inject;
import com.smartbr.dafe.util.TokenUtils;

import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import io.restassured.RestAssured;
import io.restassured.http.ContentType;
import io.restassured.http.Header;
import io.restassured.specification.RequestSpecification;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(MilkPayTestLifecycleManager.class)
public class TitleControllerTest {

    @Inject
    private TokenUtils tokenUtils;

    private String token;

    @BeforeEach
    public void genereteToken() throws Exception {
        token = tokenUtils.generateTokenTest("10564574902", "1234");
    }

    private RequestSpecification given() {
        return RestAssured.given()
                .contentType(ContentType.JSON)
                .header(new Header("Authorization", "Bearer " + token));
    }

    @Test
    @DataSet("cenario-testCase02.json")
    public void testConsultPix() {
        String result = given()
                .when().get("/milkpay-api/v1/title")
                .then()
                .statusCode(200)
                .extract().asString();

        Approvals.verifyJson(result);
    }

}