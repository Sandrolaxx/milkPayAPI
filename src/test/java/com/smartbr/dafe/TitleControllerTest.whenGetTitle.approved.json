{
  "allResultsSize": 5,
  "results": [
    {
      "amount": 80.71,
      "balance": 80.71,
      "dailyInterest": 0.30,
      "digitable": "03399000373350000000601245901010590080000008071",
      "dueDate": "2022-06-16",
      "id": 24,
      "inclusionDate": "2022-05-22T00:00:00",
      "liquidated": false,
      "nfNumber": "NF-2341252",
      "paymentType": "BOLETO"
    },
    {
      "amount": 80.71,
      "balance": 80.71,
      "dailyInterest": 0.30,
      "digitable": "03399000373350000000601245901010590080000008071",
      "dueDate": "2022-06-16",
      "id": 25,
      "inclusionDate": "2022-05-22T00:00:00",
      "liquidated": false,
      "nfNumber": "NF-2341252",
      "paymentType": "BOLETO"
    },
    {
      "amount": 80.71,
      "balance": 80.71,
      "dailyInterest": 0.30,
      "digitable": "03399000373350000000601245901010590080000008071",
      "dueDate": "2022-06-22",
      "id": 27,
      "inclusionDate": "2022-05-22T00:00:00",
      "liquidated": false,
      "nfNumber": "NF-2341252",
      "paymentType": "BOLETO"
    },
    {
      "amount": 80.71,
      "balance": 80.71,
      "dailyInterest": 0.30,
      "dueDate": "2022-06-22",
      "id": 28,
      "inclusionDate": "2022-05-22T00:00:00",
      "liquidated": false,
      "nfNumber": "NF-2341251",
      "paymentType": "PIX",
      "pixKey": "email@gmail.com"
    },
    {
      "amount": 80.71,
      "balance": 80.71,
      "dailyInterest": 0.30,
      "digitable": "03399000373350000000601245901010590080000008071",
      "dueDate": "2022-07-12",
      "id": 32,
      "inclusionDate": "2022-06-22T00:00:00",
      "liquidated": false,
      "nfNumber": "NF-2341251",
      "paymentType": "BOLETO"
    }
  ]
}